@xen-orchestra/fs/src/abstract.js

```diff
@@ -87,7 +87,9 @@ export default class RemoteHandlerAbstract {
       }
     }
     ;({ timeout: this._timeout = DEFAULT_TIMEOUT } = options)
-    maximum_operations = options.maxParallelFsOperations
+    sharedLimit = limit(
+      options.maxParallelFsOperations || DEFAULT_MAXIMUM_PARALLEL_OPERATIONS
+    )
```

Comment:
Apply the wrapper directly in the constructor.

Reply:
which constructor ?

Reply:
Here, in this constructor, apply the created limit wrapper to the methods.

Reply:
Inside the constructor, do you want me to apply the created limit wrapper to the methods ! methods are outside the constructor.

Reply:
```js
this.readFile = sharedLimit(this.readFile)
```

Reply:
Never seen this all may life, thank you :)

Reply:
What didn't you see before?

Do you understand how it works? What this does?

Reply:
I never seen the case when we change class methods in constructor. except this, I understand how it works :)